### ISSUE TITLE
cabal should warn if dependency is specified multiple times

### ISSUE THREAD

**Issue by erikd** [haskell/cabal#1107]
The buildbox-2.1.2.2 package has the following Build-Depends:

```
Build-Depends: 
    base           >= 4.4 && < 4.7,
    containers     >= 0.4 && < 0.6,
    time           >= 1.2 && < 1.5,
    directory      >= 1.1 && < 1.3,
    bytestring     >= 0.9 && < 0.11,
    mtl            == 2.1.*,
    old-locale     == 1.0.*,
    process        == 1.1.*,
    random         == 1.0.*,
    pretty         == 1.1.*,
    bytestring     == 0.10.*,
    stm            == 2.4.*
```

Note that bytestring is defined twice. cabal should detect and reject multiple definitions of the same dependency.

**Editorial by @ezyang.** Note that, with the presence of conditionals, a particular dependency can easily show up multiple times in a Cabal file. So the request here is more for a "lint" type check for when the duplication cannot possibly help.

**Comment by grayjay** [comment]
Some packages list a dependency twice to specify a range.  For example, [`hint`]  (https://hackage.haskell.org/package/hint-0.4.2.3/hint.cabal) uses `build-depends:  base >= 4, base < 5`.

**Comment by 23Skidoo** [comment]
Still, stuff like

```
    bytestring     >= 0.9 && < 0.11,
    [...]
    bytestring     == 0.10.*,
```

should probably produce a warning about redundant constraints. Fairly minor issue IMO.

**Comment by grayjay** [comment]
Yeah, it would probably be easy to distinguish between those two cases.
